### PR TITLE
Planning: fem_pos_deviation_smoother implementation complete

### PR DESCRIPTION
**Pull request by JasonZhou404**

### CODE REVIEW

File: modules/planning/math/discretized_points_smoothing/BUILD [hunk 1]
```diff
@@ -1,5 +1,7 @@
 load("//tools:cpplint.bzl", "cpplint")
 
+load("//tools:cpplint.bzl", "cpplint")
```
Comment (YajiaZhang) [inline modules/planning/math/discretized_points_smoothing/BUILD]: Remove the duplicate line

File: modules/planning/math/discretized_points_smoothing/fem_pose_deviation_smoother.cc [hunk 2]
```diff
@@ -94,8 +110,8 @@ bool FemPoseDeviationSmoother::Optimize(const OsqpSettings& solver_settings) {
   // Extract primal results
   x_.resize(num_of_points_);
   y_.resize(num_of_points_);
-  for (size_t i = 0; i < num_of_points_; ++i) {
-    size_t index = i << 1;
+  for (int i = 0; i < num_of_points_; ++i) {
+    int index = i << 1;
```
Comment (YajiaZhang) [inline modules/planning/math/discretized_points_smoothing/fem_pose_deviation_smoother.cc]: Use "plain" operations here

File: modules/planning/math/discretized_points_smoothing/fem_pose_deviation_smoother.cc [hunk 3]
```diff
@@ -112,24 +128,129 @@ bool FemPoseDeviationSmoother::Optimize(const OsqpSettings& solver_settings) {
...
+    columns[col].emplace_back(col, 5.0 * weight_fem_pose_deviation_ +
+                                       2.0 * weight_path_length_ +
+                                       weight_ref_deviation_);
+    ++col_num;
+  }
+
+  int second_point_from_last_index = num_of_points_ - 2;
+  for (int point_index = 2; point_index < second_point_from_last_index;
+       ++point_index) {
+    int col_index = point_index << 1;
```
Comment (YajiaZhang) [inline modules/planning/math/discretized_points_smoothing/fem_pose_deviation_smoother.cc]: Same here.

File: modules/planning/math/discretized_points_smoothing/fem_pose_deviation_smoother.cc [hunk 4]
```diff
@@ -112,38 +128,145 @@ bool FemPoseDeviationSmoother::Optimize(const OsqpSettings& solver_settings) {
...
                                                std::vector<c_int>* P_indices,
-                                               std::vector<c_int>* P_indptr) {}
+                                               std::vector<c_int>* P_indptr) {
+  CHECK_GT(num_of_variables_, 4);
+  // Only upper triangle is filled
+  std::vector<std::vector<std::pair<c_int, c_float>>> columns;
+  columns.resize(num_of_variables_);
+  int col_num = 0;
+
+  for (int col = 0; col < 2; ++col) {
```
Comment (xiaoxq) [inline modules/planning/math/discretized_points_smoothing/fem_pose_deviation_smoother.cc]: Better comment how you use the vectors, as there are too many magic numbers in this function. It's just not maintainable.

**Pull request merged**